lib/sha1.nit

```diff
@@ -243,35 +231,30 @@ redef class String
 
 		digested[20] = '\0';
 
-		return NativeString_to_s_with_length(digested, 20);
+		return digested;
 	`}
+end
+
+redef class String
+
+	# Computes the SHA1 of the receiver
+	#
+	# Returns a digest of 20 bytes as a NativeString,
+	# note that all the characters are not necessarily ASCII.
+	# If you want the hex string version of the digest, use
+	# sha1_hexdigest.
+	#
+	#     import base64
+	#     assert "The quick brown fox jumps over the lazy dog".sha1 == [0x2Fu8, 0xD4u8, 0xE1u8, 0xC6u8, 0x7Au8, 0x2Du8, 0x28u8, 0xFCu8, 0xEDu8, 0x84u8, 0x9Eu8, 0xE1u8, 0xBBu8, 0x76u8, 0xE7u8, 0x39u8, 0x1Bu8, 0x93u8, 0xEBu8, 0x12u8]
```

Comment:
This would be easier to read with a `Text::from_hex_digest: Bytes`, as in `... == "0x2FD4E1...u8".from_hex_digest`